lib/TopicDistribution.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2016 BigML
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+"use strict";
+
+var BigML = require('./BigML');
+var Resource = require('./Resource');
+var constants = require('./constants');
+var logger = require('./logger');
+var utils = require('./utils');
+
+function optionalCUParams(argsArray, message) {
```

Comment:
I believe it's common to all "prediction" resources. I'll check whether they are completely identical and refactor if that's the case.